dist/pathfora.js

```diff
@@ -1601,10 +1637,49 @@
         this.updateObject(widget, defaults);
         this.updateObject(widget, widget.config);
 
-        if (widget.displayConditions.showDelay) {
-          core.registerDelayedWidget(widget);
+        var displayWidget = function(widget) {
+          if (widget.displayConditions.showDelay) {
+            core.registerDelayedWidget(widget);
+          } else {
+            core.initializeWidget(widget);
+          }
+        }
+
+        if (widget.type === "message" && widget.recommend) {
+          if (widget.layout !== "slideout" && widget.layout !== "modal") {
+            throw new Error('Unsupported layout for content recommendation');
+          }
+
+          var def;
+          if (widget.content && widget.content[0]) {
+            if (widget.content[0].default)
+              def = widget.content[0];
+            else
+              throw new Error('Cannot define recommended content unless it is a default');
+          }
+
+          api.recommendContent(accountId, widget.recommend.ql.raw, function(content){
+            var con = [];
+
+            if (content) {
```

Comment:
probably can be ignored for now but i wonder how long it will be before we need to store multiple pieces of content and do a random selection on the pathfora end of recommended?